src/number_theory/number_field.lean

```diff
@@ -142,3 +144,89 @@ noncomputable def ring_of_integers_equiv : ring_of_integers ℚ ≃+* ℤ :=
 ring_of_integers.equiv ℤ
 
 end rat
+
+namespace adjoin_root
+
+section
+
+open_locale polynomial
+
+local attribute [-instance] algebra_rat
+
+/-- The quotient of `ℚ[X]` by the ideal generated by an irreducible polynomial of `ℚ[X]`
+is a number field. -/
+instance {f : ℚ[X]} [hf : irreducible f] : number_field (adjoin_root f) :=
+{ to_char_zero := char_zero_of_injective_algebra_map (algebra_map ℚ _).injective,
+  to_finite_dimensional := begin
+   let := (adjoin_root.power_basis (irreducible.ne_zero hf : f ≠ 0)),
+   convert power_basis.finite_dimensional this,
+   haveI : subsingleton (algebra ℚ (adjoin_root f)) := algebra_rat_subsingleton,
+   exact subsingleton.elim _ _,
+  end }
+
+end
+
+end adjoin_root
+
+namespace number_field.embeddings
+
+section number_field
+
+open set finite_dimensional polynomial
+
+variables {K L : Type*} [field K] [field L]
+variables [number_field K] [number_field L]  (x : K)
+
+variables {A : Type*} [field A] [char_zero A]
+
+/-- There are finitely many embeddings of a number field. -/
+noncomputable instance : fintype (K →+* A) := fintype.of_equiv (K →ₐ[ℚ] A)
+ring_hom.equiv_rat_alg_hom.symm
+
+variables [is_alg_closed A]
+
+/-- The number of embeddings of a number field is its finrank. -/
+lemma card : fintype.card (K →+* A) = finrank ℚ K :=
```

Comment:
I just realized that `K` and `A` are implicit here. When you open your next PR about this, can you make them explicit? In the next lemma `A` should be explicit.